src/DocumentDB.ChangeFeedProcessor/ChangeFeedProcessorBuilder.cs

```diff
@@ -473,10 +485,10 @@ private IPartitionManager BuildPartitionManager(ILeaseStoreManager leaseStoreMan
                     collection.PartitionKey != null &&
                     collection.PartitionKey.Paths != null &&
                     collection.PartitionKey.Paths.Count > 0;
-                if (isPartitioned && isPartitionKeyByIdRequiredIfPartitioned &&
-                    (collection.PartitionKey.Paths.Count != 1 || collection.PartitionKey.Paths[0] != "/id"))
+                if (isPartitioned &&
+                    (collection.PartitionKey.Paths.Count != 1 || !collection.PartitionKey.Paths[0].Equals($"/{this.leaseCollectionPartitionKeyPropertyName}", StringComparison.OrdinalIgnoreCase)))
                 {
-                    throw new ArgumentException("The lease collection, if partitioned, must have partition key equal to id.");
+                    throw new ArgumentException($"The lease collection, if partitioned, must have partition key equal to {this.leaseCollectionPartitionKeyPropertyName}.");
                 }
 
                 var requestOptionsFactory = isPartitioned ?
```

Comment:
PartitionedByIdCollectionRequestOptionsFactory is no longer valid name, is that right?

Reply:
will fix it